From patchwork Mon Dec  3 20:13:32 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Wei Huang <wei@redhat.com>
X-Patchwork-Id: 10710501
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 8E4BE13AF
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon,  3 Dec 2018 20:13:37 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 85A1C22ADC
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon,  3 Dec 2018 20:13:36 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 78CFB26538; Mon,  3 Dec 2018 20:13:36 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 15E0D2B3A4
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon,  3 Dec 2018 20:13:36 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1725936AbeLCUNg (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Mon, 3 Dec 2018 15:13:36 -0500
Received: from mx1.redhat.com ([209.132.183.28]:38412 "EHLO mx1.redhat.com"
        rhost-flags-OK-OK-OK-OK) by vger.kernel.org with ESMTP
        id S1725903AbeLCUNf (ORCPT <rfc822;kvm@vger.kernel.org>);
        Mon, 3 Dec 2018 15:13:35 -0500
Received: from smtp.corp.redhat.com (int-mx07.intmail.prod.int.phx2.redhat.com
 [10.5.11.22])
        (using TLSv1.2 with cipher AECDH-AES256-SHA (256/256 bits))
        (No client certificate requested)
        by mx1.redhat.com (Postfix) with ESMTPS id 7AD70308A958;
        Mon,  3 Dec 2018 20:13:34 +0000 (UTC)
Received: from weilaptop.redhat.com (ovpn-126-139.rdu2.redhat.com
 [10.10.126.139])
        by smtp.corp.redhat.com (Postfix) with ESMTP id 60FB91001F57;
        Mon,  3 Dec 2018 20:13:33 +0000 (UTC)
From: Wei Huang <wei@redhat.com>
To: kvm@vger.kernel.org
Cc: x86@kernel.org, pbonzini@redhat.com, rkrcmar@redhat.com,
        tglx@linutronix.de, mingo@redhat.com, bp@alien8.de, hpa@zytor.com,
        yunhong.jiang@intel.com
Subject: [PATCH 1/1] kvm: vmx: add cpu into VMX preemption timer bug list
Date: Mon,  3 Dec 2018 14:13:32 -0600
Message-Id: <20181203201332.20287-1-wei@redhat.com>
X-Scanned-By: MIMEDefang 2.84 on 10.5.11.22
X-Greylist: Sender IP whitelisted,
 not delayed by milter-greylist-4.5.16 (mx1.redhat.com [10.5.110.41]);
 Mon, 03 Dec 2018 20:13:34 +0000 (UTC)
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

This patch adds Intel "Xeon CPU E3-1220 V2", with CPUID.01H.EAX=0x000306A8,
into the list of known broken CPUs which fail to support VMX preemption
timer. This bug was found while running the APIC timer test of
kvm-unit-test on this specific CPU, even though the errata info can't be
located in the public domain for this CPU.

Signed-off-by: Wei Huang <wei@redhat.com>
---
 arch/x86/kvm/vmx.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/x86/kvm/vmx.c b/arch/x86/kvm/vmx.c
index 02edd9960e9d..2b90be38c02e 100644
--- a/arch/x86/kvm/vmx.c
+++ b/arch/x86/kvm/vmx.c
@@ -1764,6 +1764,8 @@ static u32 vmx_preemption_cpu_tfms[] = {
  /* 321324.pdf - AAK139 - D0 - Xeon 5500 */
  /* 320836.pdf - AAJ124 - D0 - i7-900 Extreme and i7-900 Desktop */
 0x000106A5,
+ /* Xeon E3-1220 V2 */
+0x000306A8,
 };
 
 static inline bool cpu_has_broken_vmx_preemption_timer(void)
